macros/CompatBeta.ejs

```diff
@@ -0,0 +1,603 @@
+              class="bc-level bc-level-no only-icon">
+              <span>${localize(compatStrings, 'supportsLong_no')}</span>
+            </abbr>`;
+    // We don't know when supported started
+    if (typeof(added) === 'boolean' && added) {
+      output += '?'
+    } else { // We know when
+      output += added
+    }
+    output += '&nbsp;— ';
+    // We don't know when supported ended
+    if (typeof(removed) === 'boolean' && removed) {
+      output += '?'
+    } else { // We know when
+      output += removed;
+    }
+    // removed wins over partial
+  } else if (partial) {
+    output = `<abbr title="${localize(compatStrings, 'supportsLong_partial')}"
+              class="bc-level bc-level-partial only-icon">
+              <span>${localize(compatStrings, 'supportsLong_partial')}</span>
+            </abbr>`;
+    // Display "Partial" instead of "Yes", "No", or "?" if we have no version string
+    if (typeof(added) !== 'string') {
+      output += localize(compatStrings, 'supportsShort_partial');
+    } else {
+      output += added;
+    }
+  }
+  return output;
+}
+
+/*
+Given the support information for a browser, this returns
+a CSS class to apply to the table cell.
+
+`supportData` is a (or an array of) support_statement(s)
+*/
+function getSupportClass(supportInfo) {
+  let cssClass = 'unknown';
+
+  if (Array.isArray(supportInfo)) {
+    // the first entry should be the most relevant/recent and will be treated as "the truth"
+    checkSupport(supportInfo[0].version_added,
+                 supportInfo[0].version_removed,
+                 supportInfo[0].partial_implementation);
+  } else if (supportInfo) { // there is just one support statement
+    checkSupport(supportInfo.version_added,
+                 supportInfo.version_removed,
+                 supportInfo.partial_implementation);
+  } else { // this browser has no info, it's unknown
+    return 'unknown';
+  }
+
+  function checkSupport(added, removed, partial) {
+    if (added === null) {
+      cssClass = 'unknown';
+    } else if (added) {
+      cssClass = 'yes';
+      if (removed) {
+        cssClass = 'no';
+      }
+    } else {
+      cssClass = 'no';
+    }
+    if (partial && !removed) {
+      cssClass = 'partial';
+    }
+  }
+
+  return cssClass;
+}
+
+/*
+Generate the note for a browser flag or preference
+First checks version_added and version_removed to create a string indicating when
+a preference setting is present. Then creates a (browser specific) string
+for either a preference flag or a compile flag.
+
+// TODO handle array of flags (rename flag to flags)
+// see https://github.com/mdn/browser-compat-data/issues/546
+
+`supportData` is a support_statement
+`browserId` is a compat_block browser ID
+*/
+function writeFlagsNote(supportData, browserId) {
+  let output = '';
+
+  const firefoxPrefs = 'To change preferences in Firefox, visit about:config.';
+  const chromePrefs = 'To change preferences in Chrome, visit chrome://flags.';
```

Comment:
We're localizing some strings but not most of the strings in `writeFlagsNote`. They look hard though. Not a blocker for the beta but something that should be on a to-do list somewhere.

Reply:
Added a TODO